pkg/testing/integration/program.go

```diff
@@ -178,10 +178,19 @@ type ProgramTestOptions struct {
 	RetryFailedSteps bool
 	// SkipRefresh indicates that the refresh step should be skipped entirely.
 	SkipRefresh bool
+	// SkipPreview indicates that the preview step should be skipped entirely.
+	SkipPreview bool
+	// SkipUpdate indicates that the update step should be skipped entirely.
+	SkipUpdate bool
+	// SkipExportImport skips testing that exporting and importing the stack works properly.
+	SkipExportImport bool
+	// SkipEmptyPreviewUpdate skips the no-change preview/update that is performed that validates
+	// that no changes happen.
+	SkipEmptyPreviewUpdate bool
 	// SkipStackRemoval indicates that the stack should not be removed. (And so the test's results could be inspected
 	// in the Pulumi Service after the test has completed.)
 	SkipStackRemoval bool
-	// Quick can be set to true to run a "quick" test that skips any non-essential steps (e.g., empty updates).
+	// Quick implies SkipPreview, SkipExportImport and SkipEmptyPreviewUpdate
 	Quick bool
```

Comment:
we coudl also extend this to encompass skipping refresh.  However, that would be an actual change in behavior from what we do today.